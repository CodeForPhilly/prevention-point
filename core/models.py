from django.db import models
from django.utils import timezone
from enum import Enum
from core.participants.models import Participant
class Medication(models.Model):
    participant = models.ForeignKey(Participant, on_delete=models.CASCADE)
    medical_delivery = models.CharField(max_length=100)
    medication_name = models.CharField(max_length=100)
    ingestion_frequency = models.IntegerField()

class UrineDrugScreen(models.Model):
    participant = models.ForeignKey(Participant, on_delete=models.CASCADE)
    date_of_test = models.DateField()
    uds_temp = models.CharField(max_length=100, verbose_name="Urine Temperature")
    # TODO: does this belong here? if so, is it a Bool for if a PT was taken, or Bool for if pregnant?
    pregnancy_test = models.BooleanField(default=False)
    opiates = models.BooleanField(default=False)
    fentanyl = models.BooleanField(default=False)
    bup = models.BooleanField(default=False, verbose_name="Buprenorphine")
    coc = models.BooleanField(default=False, verbose_name="Cocaine")
    amp = models.BooleanField(default=False, verbose_name = "Amphetamine")
    m_amp = models.BooleanField(default=False, verbose_name = "Methamphetamine")
    thc = models.BooleanField(default=False, verbose_name = "THC")
    mtd = models.BooleanField(default=False, verbose_name = "Methadone")
    pcp = models.BooleanField(default=False, verbose_name = "PCP")
    bar = models.BooleanField(default=False, verbose_name = "Barbiturates")
    bzo = models.BooleanField(default=False, verbose_name = "Benzodiazepines")
    tca = models.BooleanField(default=False, verbose_name = "Tricyclic Antidepressants")
    oxy = models.BooleanField(default=False, verbose_name = "Oxycodone")

    def __str__(self):
        return 'UDS #%i for Participant #%i' % (self.id, self.participant_id)

class EmployeeRole(models.Model):
    role_value = models.CharField(max_length=30)

    def __str__(self):
        return '%s' % (self.role_value)

# TODO: seed with front desk, case manager, admin

class Employee(models.Model):
    first_name = models.CharField(max_length=50)
    last_name = models.CharField(max_length=100)
    role = models.ForeignKey(EmployeeRole, on_delete=models.CASCADE)

    def __str__(self):
        return '%s %s %s' % (self.first_name, self.last_name, self.role)

class CaseManagement(models.Model):
    participant = models.ForeignKey(Participant, on_delete=models.CASCADE)
    crs_seen = models.BooleanField(default=False)

class Appointment(models.Model):
    participant = models.ForeignKey(Participant, on_delete=models.CASCADE)

    cm_employee = models.ForeignKey(
            Employee, related_name='cm_%(class)s', on_delete=models.CASCADE
            )

    crs_employee = models.ForeignKey(
            Employee, related_name='crs_%(class)s', on_delete=models.CASCADE
            )

    appointment_timestamp = models.DateTimeField()
    crs_timestamp = models.DateTimeField()
    crs_seen = models.BooleanField(default=False)
    cm_timestamp = models.DateTimeField()
    cm_seen = models.BooleanField(default=False)
    hcv_status = models.BooleanField(default=False)

# TODO: we probably want to move the HCV data into its own table(s), similar to how UDS works

class BehavioralHealthNotes(models.Model):
    participant = models.ForeignKey(Participant, on_delete=models.CASCADE)
    employee = models.ForeignKey(Employee, on_delete=models.CASCADE)
    note_timestamp = models.DateTimeField()
    behavior_note = models.TextField()

class HCVNotes(models.Model):
    participant = models.ForeignKey(Appointment, on_delete=models.CASCADE)
    employee = models.ForeignKey(Employee, on_delete=models.CASCADE)
    note_timestamp = models.DateTimeField()
    hcv_note = models.TextField()

class Visit(models.Model):
    participant = models.ForeignKey(Participant, on_delete=models.CASCADE)
    created_at = models.DateTimeField(default=timezone.now)

class ServiceEvent(models.Model):
    class ServiceEventPurpose(Enum):
        ARRIVED = 'arrived'
        CALLED = 'called'
        SEEN = 'seen'
        LEFT = 'left'

    visit = models.ForeignKey(Visit, on_delete=models.CASCADE)
    # service = models.ForeignKey(Service, on_delete=models.CASCADE)
    purpose = models.CharField(
        max_length=10,
        choices=[(tag.name, tag.value) for tag in ServiceEventPurpose]
    )
