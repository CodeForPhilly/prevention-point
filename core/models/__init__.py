from .insurer import *
from .participant import *
from .program import *
from .service import *
from .program_service_map import *
from .visit import *
from .hcv_notes import *
from .medication import *
from .appointment import *
from .case_management import *
from .front_desk_event import *
from .urine_drug_screen import *
from .program_availability import *
from .behavioral_health_notes import *