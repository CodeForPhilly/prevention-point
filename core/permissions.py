from django.contrib.auth.models import User, Group
from rest_framework.permissions import DjangoModelPermissions, BasePermission

FRONT_DESK = 'front_desk'
CASE_MANAGER = 'case_manager'
ADMIN = 'admin'

class DjangoModelPermissions(DjangoModelPermissions):
    """
    The request is authenticated using `django.contrib.auth` permissions.
    See: https://docs.djangoproject.com/en/dev/topics/auth/#permissions
    It ensures that the user is authenticated, and has the appropriate
    `add`/`change`/`delete` permissions on the model.
    This permission can only be applied against view classes that
    provide a `.queryset` attribute.


    Prevention Point Note: We are extending this Class as to include permission checks on GET
    requests. You MUST use this class to protect GET requests to endpoints. The DRF class this 
    ineherits from does not protect GET requests
    """
    perms_map = {
        'OPTIONS': [],
        'HEAD': [],
        'GET': ['%(app_label)s.view_%(model_name)s'],
        'POST': ['%(app_label)s.add_%(model_name)s'],
        'PUT': ['%(app_label)s.change_%(model_name)s'],
        'PATCH': ['%(app_label)s.change_%(model_name)s'],
        'DELETE': ['%(app_label)s.delete_%(model_name)s'],
    
    }
