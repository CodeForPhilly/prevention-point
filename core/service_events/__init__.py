from .serializer import *
from .views import *
