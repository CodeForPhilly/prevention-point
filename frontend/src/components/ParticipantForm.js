import React, { Fragment } from "react"
import PropTypes from "prop-types"
import { makeStyles } from "@material-ui/styles"
import Grid from "@material-ui/core/Grid"
import Radio from "@material-ui/core/Radio"
import Select from "@material-ui/core/Select"
import MenuItem from "@material-ui/core/MenuItem"
import TextField from "@material-ui/core/TextField"
import FormLabel from "@material-ui/core/FormLabel"
import InputLabel from "@material-ui/core/InputLabel"
import RadioGroup from "@material-ui/core/RadioGroup"
import FormControl from "@material-ui/core/FormControl"
import FormControlLabel from "@material-ui/core/FormControlLabel"
import PrevPointInput from "./Input/PrevPointInput"
import { PrevPointHeading } from "./Typography"
import {
  PARTICIPANT_GENDER_OPTIONS,
  PARTICIPANT_RACE_OPTIONS,
} from "../constants"

const useStyles = makeStyles(() => ({
  root: {
    "&$disabled, & $disabled": {
      color: "rgba(0, 0, 0, 0.85)",
    },
  },
  disabled: {},
}))

const ParticipantForm = ({
  participantInfo,
  insurers,
  isDisabled,
  handleParticipantChange,
}) => {
  const classes = useStyles()
  return (
    <Fragment>
      <Grid item xs={12}>
        <PrevPointHeading>1. Participant Information</PrevPointHeading>
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled}>
          <InputLabel htmlFor="firstName">First Name</InputLabel>
          <PrevPointInput
            id="firstName"
            name="first_name"
            value={participantInfo.first_name}
            onChange={e => handleParticipantChange(e.target)}
            required
            classes={{ root: classes.root, disabled: classes.disabled }}
          />
        </FormControl>
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled}>
          <InputLabel htmlFor="lastName">Last Name</InputLabel>
          <PrevPointInput
            id="lastName"
            name="last_name"
            value={participantInfo.last_name}
            onChange={e => handleParticipantChange(e.target)}
            required
            classes={{ root: classes.root, disabled: classes.disabled }}
          />
        </FormControl>
      </Grid>
      <Grid item xs={12} md={6}>
        <TextField
          label="Date of Birth"
          name="date_of_birth"
          type="date"
          required
          disabled={isDisabled}
          fullWidth
          value={participantInfo.date_of_birth}
          onChange={e => handleParticipantChange(e.target)}
          InputLabelProps={{
            shrink: true,
          }}
          InputProps={{
            classes: { root: classes.root, disabled: classes.disabled },
          }}
        />
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled}>
          <InputLabel htmlFor="uuid">UUID</InputLabel>
          <PrevPointInput
            id="uuid"
            name="pp_id"
            value={participantInfo.pp_id}
            onChange={e => handleParticipantChange(e.target)}
            required
            classes={{ root: classes.root, disabled: classes.disabled }}
          />
        </FormControl>
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled}>
          <InputLabel htmlFor="sep_id">SEP ID</InputLabel>
          <PrevPointInput
            id="sep_id"
            name="sep_id"
            value={participantInfo.sep_id}
            onChange={e => handleParticipantChange(e.target)}
            required
            classes={{ root: classes.root, disabled: classes.disabled }}
          />
        </FormControl>
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled}>
          <InputLabel htmlFor="maiden_name">Maiden Name</InputLabel>
          <PrevPointInput
            id="maiden_name"
            name="maiden_name"
            value={participantInfo.maiden_name}
            onChange={e => handleParticipantChange(e.target)}
            required
            classes={{ root: classes.root, disabled: classes.disabled }}
          />
        </FormControl>
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled}>
          <InputLabel id="race">Select Race</InputLabel>
          <Select
            required
            name="race"
            value={participantInfo.race}
            onChange={e => handleParticipantChange(e.target)}
            labelId="race"
            classes={{ root: classes.root, disabled: classes.disabled }}
          >
            {PARTICIPANT_RACE_OPTIONS.map((race, index) => (
              <MenuItem key={index} value={race.value}>
                {race.name}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled}>
          <InputLabel id="participant-gender">Select Gender</InputLabel>
          <Select
            required
            name="gender"
            value={participantInfo.gender}
            onChange={e => handleParticipantChange(e.target)}
            labelId="participant-gender"
            classes={{ root: classes.root, disabled: classes.disabled }}
          >
            {PARTICIPANT_GENDER_OPTIONS.map((gender, index) => (
              <MenuItem key={index} value={gender.value}>
                {gender.name}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </Grid>
      <Grid item xs={12} md={6}>
        <FormControl disabled={isDisabled} component="fieldset">
          <FormLabel component="legend">Has Insurance?</FormLabel>
          <RadioGroup
            aria-label="insurance"
            name="is_insured"
            value={participantInfo.is_insured}
            onChange={e => handleParticipantChange(e.target)}
            style={{ display: "inline" }}
            classes={{ root: classes.root }}
          >
            <FormControlLabel
              value={true}
              control={<Radio />}
              label="Yes"
              classes={{ root: classes.root, disabled: classes.disabled }}
            />
            <FormControlLabel
              value={false}
              control={<Radio />}
              label="No"
              classes={{ root: classes.root, disabled: classes.disabled }}
            />
          </RadioGroup>
        </FormControl>
      </Grid>
      {participantInfo.is_insured && (
        <Grid item xs={12} md={6}>
          <FormControl>
            <InputLabel id="insurance-select">Select Insurance</InputLabel>
            <Select
              name="insurer"
              value={participantInfo.insurer}
              onChange={e => handleParticipantChange(e.target)}
              labelId="insurance-select"
              classes={{ root: classes.root, disabled: classes.disabled }}
            >
              {insurers.map(company => (
                <MenuItem key={company.id} value={company.id}>
                  {company.name || ""}
                </MenuItem>
              ))}
            </Select>
          </FormControl>
        </Grid>
      )}
    </Fragment>
  )
}

ParticipantForm.propTypes = {
  participantInfo: PropTypes.object,
  isDisabled: PropTypes.bool,
  insurers: PropTypes.array,
  handleParticipantChange: PropTypes.func,
}

ParticipantForm.defaultProps = {
  isDisabled: false,
}

export default ParticipantForm
