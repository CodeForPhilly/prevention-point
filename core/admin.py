from django.contrib import admin
from core.models import Participant, Medication, CaseManagement, Appointment, Employee, BehavioralHealthNotes
from core.service_events.models import ServiceEvent
from core.urine_drug_screens.models import UrineDrugScreen

# Register your models here.
admin.site.register(Medication)
admin.site.register(UrineDrugScreen)
admin.site.register(CaseManagement)
admin.site.register(Appointment)
admin.site.register(Employee)
admin.site.register(BehavioralHealthNotes)
admin.site.register(ServiceEvent)

class BaseInline(admin.StackedInline):
    extra = 0

class UrineDrugScreenInline(BaseInline):
    model = UrineDrugScreen

class AppointmentInline(BaseInline):
    model = Appointment

class MedicationInline(BaseInline):
    model = Medication

class CaseManagement(BaseInline):
    model = CaseManagement

class ParticipantAdmin(admin.ModelAdmin):
    inlines = [
            AppointmentInline,
            MedicationInline,
            CaseManagement,
            UrineDrugScreenInline,
            ]

admin.site.register(Participant, ParticipantAdmin)
