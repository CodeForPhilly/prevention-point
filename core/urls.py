"""preventionpoint URL Configuration

The `urlpatterns` list routes URLs to views. For more information please see:
    https://docs.djangoproject.com/en/2.1/topics/http/urls/
Examples:
Function views
    1. Add an import:  from my_app import views
    2. Add a URL to urlpatterns:  path('', views.home, name='home')
Class-based views
    1. Add an import:  from other_app.views import Home
    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
Including another URLconf
    1. Import the include() function: from django.urls import include, path
    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
"""
from django.contrib import admin
from django.urls import include, path
from rest_framework import routers
from rest_framework_swagger.views import get_swagger_view
from rest_framework_simplejwt import views as jwt_views

from core.users import views as user_views
from core.urine_drug_screens import views as uds_views
from core.participants import views as participant_views
from core.visits import views as visits_views
from core.front_desk_events import views as front_events_views
from core.programs import views as programs_views
from core.services import views as services_views
from core.queue import views as queues_views
from core.program_availability import views as program_availability_views
from core.program_service_map import views as program_service_map_views
from core.insurers import views as insurer_views
from core.medications import views as medications_views

admin.site.site_header = "Prevention Point Philadelphia"

router = routers.DefaultRouter()
router.register(r"users", user_views.UserViewSet)
router.register(r"groups", user_views.GroupViewSet)
router.register(r"visits", visits_views.VisitViewSet)
router.register(r"uds", uds_views.UrineDrugScreenViewSet)
router.register(r"medications", medications_views.MedicationViewSet)
router.register(r"participants", participant_views.ParticipantViewSet)
router.register(r"front-desk-events", front_events_views.FrontDeskEventViewSet)
router.register(r"programs", programs_views.ProgramViewSet)
router.register(r"program-service-map", program_service_map_views.ProgramServiceMapView)
router.register(r"services", services_views.ServiceViewSet)
router.register(r"insurers", insurer_views.InsurerViewSet)


schema_view = get_swagger_view(title="PreventionPoint API")

urlpatterns = [
    path("api/", include(router.urls)),
    path(
        "api/token/", jwt_views.TokenObtainPairView.as_view(), name="token_obtain_pair"
    ),
    path(
        "api/token/refresh/", jwt_views.TokenRefreshView.as_view(), name="token_refresh"
    ),
    path("api/token/verify/", jwt_views.TokenVerifyView.as_view(), name="token_verify"),
    path(
        "api/programs/<int:program_id>/program-availability/",
        program_availability_views.ProgramAvailabilityViewSet.as_view({"get": "list"}),
    ),
    path(
        "api/programs/<int:program_id>/program-availability/<int:pk>/",
        program_availability_views.ProgramAvailabilityViewSet.as_view(
            {"put": "update"}
        ),
    ),
    path(
        "api/programs/<int:program_id>/queue/",
        queues_views.QueueViewSet.as_view({"get": "retrieve"}),
    ),
    # above only 'registers' the retrieve path. to use other CRUD actions, must pass that explicitly to .as_view in another register
    path("swagger/", schema_view),
    path("admin/", admin.site.urls),
]
