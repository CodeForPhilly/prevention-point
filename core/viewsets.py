from rest_framework import generics, viewsets
from rest_framework.permissions import IsAuthenticated
from core.permissions import DjangoModelPermissions

class ModelViewSet(viewsets.ModelViewSet):
    permission_classes = [DjangoModelPermissions, IsAuthenticated]
