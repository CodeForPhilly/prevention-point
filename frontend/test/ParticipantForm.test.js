/* eslint-disable quotes */
import React from "react"
import { shallow, configure } from "enzyme"
import { createMount } from "@material-ui/core/test-utils"
import Adapter from "enzyme-adapter-react-16"
import { format } from "date-fns"
import ParticipantForm from "../src/components/ParticipantForm"

configure({ adapter: new Adapter() })

describe("<ParticipantForm />", () => {
  let wrapper
  let mount

  // Create initial props that get passed into the component
  const initialProps = {
    participantInfo: {
      id: null,
      first_name: "foobar",
      last_name: "",
      last_four_ssn: 0,
      date_of_birth: "",
      start_date: format(new Date(), "yyyy-MM-dd"),
      pp_id: "",
      race: "",
      gender: "",
      is_insured: false,
      insuranceType: "",
      insurer: "",
    },
    insurers: [{ id: 1, name: "insurer" }],
    setRace: () => {},
    setPPId: () => {},
    setGender: () => {},
    setInsurer: () => {},
    setLastName: () => {},
    setIsInsured: () => {},
    setFirstName: () => {},
    setDateOfBirth: () => {},
  }

  // Unit testing
  describe("Unit tests", () => {
    // what to do before each test
    beforeEach(() => {
      // Render the login form component, pass in props. (Shallow method renders the component without its children, good for unit tests.)
      mount = createMount()
      wrapper = shallow(<ParticipantForm {...initialProps} />)
    })

    // what to do after each test
    afterEach(() => {
      jest.clearAllMocks()
    })

    it("should work", () => {
      wrapper = mount(<ParticipantForm {...initialProps} />)
    })
    // UI Integrity test
    it("should match the snapshot", () => {
      // snapshots are text references of the html of the rendered component.
      expect(wrapper.html()).toMatchSnapshot()
    })

    it("should have a firstname input", () => {
      let input = wrapper.find('[name="first_name"]')
      expect(input.length).toEqual(1)
    })

    it("should have a last name input", () => {
      let input = wrapper.find('[name="last_name"]')
      expect(input.length).toEqual(1)
    })

    it("should have a date of birth input", () => {
      let input = wrapper.find('[name="date_of_birth"]')
      expect(input.length).toEqual(1)
    })

    it("should have a has insurance input", () => {
      let input = wrapper.find('[name="is_insured"]')
      expect(input.length).toEqual(1)
    })

    it("should not have an insurer input when is_insured is false", () => {
      let input = wrapper.find('[name="insurer"]')
      expect(input.length).toEqual(0)
    })

    it("should have an insurer input when is_insured is true", () => {
      wrapper.setProps({ participantInfo: { is_insured: true } })
      let input = wrapper.find('[name="insurer"]')
      expect(input.length).toEqual(1)
    })

    it("should have a gender input", () => {
      let input = wrapper.find('[name="gender"]')
      expect(input.length).toEqual(1)
    })

    it("should have a race input", () => {
      let input = wrapper.find('[name="race"]')
      expect(input.length).toEqual(1)
    })
  })
})
