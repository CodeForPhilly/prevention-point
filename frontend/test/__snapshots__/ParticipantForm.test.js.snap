// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipantForm /> Unit tests should match the snapshot 1`] = `"<div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><h2 class=\\"MuiTypography-root MuiTypography-h2 MuiTypography-displayBlock\\" aria-label=\\"heading\\">1. Participant Information</h2></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"firstName\\">First Name</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" required=\\"\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"firstName\\" name=\\"first_name\\" required=\\"\\" value=\\"foobar\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"lastName\\">Last Name</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" required=\\"\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"lastName\\" name=\\"last_name\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required\\" data-shrink=\\"true\\">Date of Birth<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\">â€‰*</span></label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"date\\" aria-invalid=\\"false\\" name=\\"date_of_birth\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"uuid\\">UUID</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" required=\\"\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"uuid\\" name=\\"pp_id\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" id=\\"race\\">Select Race</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" required=\\"\\"><div class=\\"MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\" aria-labelledby=\\"race mui-component-select-race\\" id=\\"mui-component-select-race\\"><span>&#8203;</span></div><input type=\\"hidden\\" value=\\"\\" name=\\"race\\"/><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" id=\\"participant-gender\\">Select Gender</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\" required=\\"\\"><div class=\\"MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\" aria-labelledby=\\"participant-gender mui-component-select-gender\\" id=\\"mui-component-select-gender\\"><span>&#8203;</span></div><input type=\\"hidden\\" value=\\"\\" name=\\"gender\\"/><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><fieldset class=\\"MuiFormControl-root\\"><legend class=\\"MuiFormLabel-root\\">Has Insurance?</legend><div class=\\"MuiFormGroup-root\\" role=\\"radiogroup\\" aria-label=\\"insurance\\" style=\\"display:inline\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-227 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"radio\\" class=\\"PrivateSwitchBase-input-230\\" name=\\"is_insured\\" value=\\"true\\"/><div class=\\"PrivateRadioButtonIcon-root-243\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg><svg class=\\"MuiSvgIcon-root PrivateRadioButtonIcon-layer-244\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z\\"></path></svg></div></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">Yes</span></label><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-227 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-228 Mui-checked MuiIconButton-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"radio\\" checked=\\"\\" class=\\"PrivateSwitchBase-input-230\\" name=\\"is_insured\\" value=\\"false\\"/><div class=\\"PrivateRadioButtonIcon-root-243 PrivateRadioButtonIcon-checked-245\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg><svg class=\\"MuiSvgIcon-root PrivateRadioButtonIcon-layer-244\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z\\"></path></svg></div></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">No</span></label></div></fieldset></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\"><div class=\\"MuiFormControl-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" id=\\"insurance-select\\">Select Insurance</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><div class=\\"MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\" aria-labelledby=\\"insurance-select mui-component-select-insurer\\" id=\\"mui-component-select-insurer\\"><span>&#8203;</span></div><input type=\\"hidden\\" value=\\"\\" name=\\"insurer\\"/><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div></div>"`;
